src/util/OrmUtils.ts

```diff
@@ -177,7 +177,9 @@ export class OrmUtils {
         if (x === y)
             return true;
 
-        if (x.equals instanceof Function && x.equals(y))
+        // Check if both argument is Buffer
+        // Buffer.equals Argument must be a Buffer
+        if (Buffer.isBuffer(x) && Buffer.isBuffer(y) && x.equals(y))
```

Comment:
Does anyone know x and y have `.equals` method expect Buffer?

Reply:
isn't `.equals` come from buffer? If yes then maybe we can check if x and y are both instances of objects?

Reply:
equals come from Buffer.
Object.equals is not defined
```TypeScript
const a = new Buffer([]);
const b = {}
a.equals(b); // TypeError: Argument must be a Buffer
b.equals // undefined
```
So you must check instances are both Buffer or not.

Reply:
ah sorry looks like equals in in a Function. Why did you remove previous line?

```
if (x.equals instanceof Function && x.equals(y))
```

?

Reply:
Because it seems that line is just for Buffer class.
So I fixed that line(not remove).

Reply:
hmmm I wouldn't assume that. For example mongodb's ObjectId has equals method as well. I don't remember why exactly we need this (my bad on missing comment) but we need to put it back until we figure out that we really don't need it.

Reply:
Also please add a test that **fails** if someday somebody remove lines of code you have added.

Reply:
yes, I think we need more unit tests to describe logic.
ok, I'll prepare unit test for this case.